File: Week_02/cpp/src/_389_FindTheDifference.cpp
#include <string>
using namespace std;
class MySolution
{
public:
    char findTheDifference(string s, string t)
    {
        int freq[26] = {0};
        for (char it : s)
        {
            freq[it - 'a']++;
        }
        for (char it : t)
        {
            freq[it - 'a']--;
        }
        for (int i = 0; i < 26; i++)
        {
            if (freq[i])
                return i + 'a';
        }
        return 0;
    }
};
class UsingMap
{
public:
    char findTheDifference(string s, string t)
    {
        int freq[26] = {0};
        for (auto &it : s)
        {
            freq[it - 'a']++;
        }
        for (auto &it : t)
        {
            if (--freq[it - 'a'] < 0)
                return it;
        }
        return ' ';
    }
};
class UsingSum
{
public:
    char findTheDifference(string s, string t)
    {
        int ret = 0;
        for (auto &it : t)
        {
            ret += it;
        }
        for (auto &it : s)
        {
            ret -= it;
        }
        return ret;
    }
};
class UsingBitwise
{
public:
    char findTheDifference(string s, string t)
    {
        int ret = 0;
        for (auto &it : s)
        {
            ret ^= it;
        }
        for (auto &it : t)
        {
            ret ^= it;
        }
        return ret;
    }
};

File: Week_02/kotlin/src/_389_FindTheDifference.kt
class _389_FindTheDifference_ {
    class UsingMap {
        fun findTheDifference(s: String, t: String): Char {
            val freq = IntArray(26)
            s.forEach { freq[it - 'a']++ }
            t.forEach {
                if (--freq[it - 'a'] < 0) {
                    return it
                }
            }
            return ' '
        }
    }
    class UsingSum {
        fun findTheDifference(s: String, t: String): Char {
            var ret = 0
            t.forEach { ret += it.toInt() }
            s.forEach { ret -= it.toInt() }
            return ret.toChar()
        }
    }
    class UsingXor {
        fun findTheDifference(s: String, t: String): Char {
            var ret = 0
            s.forEach { ret = ret xor it.toInt() }
            t.forEach { ret = ret xor it.toInt() }
            return ret.toChar()
        }
    }
}